File: cses/resturant.py
def solve():
    n = int(input())
    timing = []
    for _ in range(n):
        time = list(map(int, input().split()))
        timing.append(time)

    timing.sort(key=lambda t: t[0])
    # print(timing)

    max_cust, cust = 0, 0
    last_cust = -1
    for time in timing:
        if time[0] <= last_cust:
            cust += 1
            last_cust = min(time[1], last_cust)
        else:
            cust = 1
            last_cust = time[1]
        max_cust = max(max_cust, cust)
    print(max_cust)


def main():
    solve()


if __name__ == "__main__":
    main()

File: cses/resturant.cpp
#include <algorithm>
#include <iostream>
#include <map>
using namespace std;

#define ll long long

void solve() {
    int n;
    cin >> n;
    map<ll, int> mp;

    for (int i = 0; i < n; ++i) {
        ll arr, dep;
        cin >> arr >> dep;

        mp[arr] = 1;
        mp[dep] = -1;
    }

    ll present = 0, maxCust = 0;
    for (auto [time, cust] : mp) {
        present += cust;
        maxCust = max(maxCust, present);
    }
    cout << maxCust << '\n';
}

int main() {
    solve();
    return 0;
}
